openassessment/xblock/data_layer/serializers.py

```diff
@@ -0,0 +1,201 @@
+
+    text_response_config = TextResponseConfigSerializer(source="*")
+    file_response_config = FileResponseConfigSerializer(source="*")
+
+    teams_config = TeamsConfigSerializer(source="*")
+
+
+class RubricFeedbackConfigSerializer(Serializer):
+    description = CharField(source="rubric_feedback_prompt")  # is this this field?
+    default_text = CharField(source="rubric_feedback_default_text")
+
+
+class RubricCriterionOptionSerializer(Serializer):
+    name = CharField()
+    label = CharField()
+    points = IntegerField()
+    description = CharField(source="explanation")
+
+
+class RubricCriterionSerializer(Serializer):
+    name = CharField(source="label")
+    description = CharField(source="prompt")
+    feedback_enabled = SerializerMethodField()
+    feedback_required = SerializerMethodField()
+    options = RubricCriterionOptionSerializer(many=True)
+
+    @staticmethod
+    def _feedback(criterion):
+        return criterion.get("feedback", "disabled")
+
+    def get_feedback_enabled(self, criterion):
+        return self._feedback(criterion) != "disabled"
+
+    def get_feedback_required(self, criterion):
+        return self._feedback(criterion) == "required"
+
+
+class RubricConfigSerializer(Serializer):
+    show_during_response = BooleanField(source="show_rubric_during_response")
+    feedback_config = RubricFeedbackConfigSerializer(source="*")
+    criteria = RubricCriterionSerializer(
+        many=True, source="rubric_criteria_with_labels"
+    )
+
+
+class RequiredMixin(Serializer):
+    required = BooleanField(default=True)
+
+
+class StartEndMixin(Serializer):
+    start = DateTimeField()
+    due = DateTimeField()
+
+
+class TrainingSettingsSerializer(RequiredMixin, Serializer):
+    pass
+
+
+class PeerSettingsSerializer(RequiredMixin, StartEndMixin, Serializer):
+    min_number_to_grade = IntegerField(source="must_grade")
+    min_number_to_be_graded_by = IntegerField(source="must_be_graded_by")
+    flexible_grading = BooleanField(source="enable_flexible_grading", required=False)
+
+
+class SelfSettingsSerializer(RequiredMixin, Serializer):
+    pass
+
+
+class StaffSettingsSerializer(RequiredMixin, Serializer):
+    pass
+
+
+class AssessmentStepsSettingsSerializer(Serializer):
+    training_step = SerializerMethodField(label="training")
+    peer_step = SerializerMethodField(label="peer")
+    self_step = SerializerMethodField(label="self")
+    staff_step = SerializerMethodField(label="staff")
+
+    def _get_step(self, instance, step_name):
+        """Get the assessment step config for a given step_name"""
+        for step in instance.rubric_assessments:
+            if step["name"] == step_name:
+                return step
+        return None
+
+    def get_training_step(self, instance):
+        """Get the training step configuration"""
+        training_step = self._get_step(instance, "student-training")
+        return TrainingSettingsSerializer(training_step).data or {}
```

Comment:
I wonder if it would be easy to do this as a custom Field type to remove the duplication here

Reply:
@jansenk , how's this new version look? https://github.com/openedx/edx-ora2/pull/2002/commits/e6c6de221dcc11832b3deab2e6b6221cacae6fa2